engine/table/src/main/java/io/deephaven/engine/table/impl/by/AggregationProcessor.java

```diff
@@ -707,6 +715,34 @@ public void visit(@NotNull final Partition partition) {
                     groupByColumnNames));
         }
 
+        @Override
+        public void visit(@NotNull final Formula formula) {
+            final SelectColumn selectColumn =
+                    SelectColumn.of(Selectable.of(formula.column(), RawString.of(formula.formula())));
+
+            // Get or create a column definition map composed of vectors of the original column types.
+            if (vectorColumnNameMap == null) {
+                vectorColumnNameMap = new HashMap<>();
+                table.getColumnSourceMap().forEach((key, value) -> {
+                    final ColumnDefinition<?> columnDef = ColumnDefinition.fromGenericType(
+                            key, VectorFactory.forElementType(value.getType()).vectorType());
+                    vectorColumnNameMap.put(key, columnDef);
+                });
```

Comment:
To confirm, there's no risk of dependencies between these columns such that we would want vectorColumnMap to be a LinkedHashMap?

Reply:
No, this is created only for an input to `SelectColumn#initDef`.  The column will eventually operate on vectors of values that are independent.